[docs] Smart driver topology fallback and server refresh (#15906)
yugabyte/yugabyte-db: docs/content/preview/drivers-orms/go/yb-pgx.md

</ul>

> The [YugabyteDB PGX smart driver](https://pkg.go.dev/github.com/yugabyte/pgx) is a Go driver for [YSQL](/preview/api/ysql/) based on [jackc/pgx](https://github.com/jackc/pgx/), with a additional [connection load balancing](../../smart-drivers/) features.

Reviewer: ```suggestion
The [YugabyteDB PGX smart driver](https://pkg.go.dev/github.com/yugabyte/pgx) is a Go driver for [YSQL](/preview/api/ysql/) based on [jackc/pgx](https://github.com/jackc/pgx/), with additional [connection load balancing](../../smart-drivers/) features.
```

> The driver makes an initial connection to the first contact point provided by the application to discover all the nodes in the cluster and, if required, refreshes the list of live endpoints with every new connection attempt. The refresh is triggered if stale information (by default, older than 5 minutes) is discovered.

Reviewer: ```suggestion
The driver makes an initial connection to the first contact point provided by the application to discover all the nodes in the cluster. If the driver discovers stale information (by default, older than 5 minutes), it refreshes the list of live endpoints with every new connection attempt.
```

> The [YugabyteDB PGX smart driver](https://pkg.go.dev/github.com/yugabyte/pgx) is a distributed Go driver for [YSQL](/preview/api/ysql/) based on [jackc/pgx](https://github.com/jackc/pgx/), with a additional [connection load balancing](../../smart-drivers/) features.

Reviewer 2: Not sure we use word `distributed` for a driver.

| password | User password | yugabyte
| dbname | Database name | yugabyte
| `load_balance` | [Uniform load balancing](../../smart-drivers/#cluster-aware-connection-load-balancing) | Defaults to upstream driver behavior unless set to 'true'

> | `yb_servers_refresh_interval` | Interval in seconds to refresh the servers list | 300

Reviewer 2: We can add that this is applicable only when load-balance is set to true.

| `yb_servers_refresh_interval` | If `load_balance` is true, the interval in seconds to refresh the servers list | 300
| `topology_keys` | [Topology-aware load balancing](../../smart-drivers/#topology-aware-connection-load-balancing) | If `load_balance` is true, uses uniform load balancing unless set to comma-separated geo-locations in the form `cloud.region.zone`.

The following is an example connection string for connecting to YugabyteDB with uniform load balancing:

```sh
postgres://username:password@localhost:5433/database_name?load_balance=true& \
    yb_servers_refresh_interval=240
```

The following is a code snippet for connecting to YugabyteDB using the connection parameters:

```go
baseUrl := fmt.Sprintf("postgres://%s:%s@%s:%d/%s",
                    user, password, host, port, dbname)
url := fmt.Sprintf("%s?load_balance=true&yb_servers_refresh_interval=240", baseUrl)
conn, err := pgx.Connect(context.Background(), url)
```

The following is an example connection string for connecting to YugabyteDB with topology-aware load balancing:

```sh

> postgres://username:password@localhost:5433/database_name?load_balance=true& \

Reviewer 2: If this is expected to be copied as-is, then `\` won't be useful in a Go app. Instead you can either put it in a single line or use `+`
```
url := "postgres://username:password@localhost:5433/database_name?load_balance=true&" +
    "topology_keys=cloud1.region1.zone1,cloud1.region1.zone2"
```

postgres://username:password@host1:5433,host2:5433,host3:5433/database_name?load_balance=true
```

> To query the cluster for a list of available hosts, use the `yb_servers()` function.

Reviewer 2: We don't expect users to query the yb_servers() function. It is the driver who does it. This statement may confuse readers. Same holds true on other language pages.
